bft/testers/snowflake/runner.py

```diff
@@ -0,0 +1,184 @@
+import datetime
+import math
+import os
+import yaml
+from typing import Dict, NamedTuple
+
+from snowflake.connector import connect
+from snowflake.connector.errors import Error
+
+from bft.cases.runner import SqlCaseResult, SqlCaseRunner
+from bft.cases.types import Case
+from bft.dialects.types import SqlMapping
+
+type_map = {
+    "i8": "TINYINT",
+    "i16": "SMALLINT",
+    "i32": "INTEGER",
+    "i64": "BIGINT",
+    "fp32": "REAL",
+    "fp64": "DOUBLE",
+    "boolean": "BOOLEAN",
+    "string": "VARCHAR",
+    "date": "DATE",
+    "time": "TIME",
+    "timestamp": "TIMESTAMP",
+    "timestamp_tz": "TIMESTAMPTZ",
+    "interval": "INTERVAL",
+}
+
+
+def type_to_snowflake_type(type: str):
+    if type not in type_map:
+        raise Exception(f"Unrecognized type: {type}")
+    return type_map[type]
+
+
+def literal_to_str(lit: str | int | float):
+    if lit is None:
+        return "null"
+    elif lit in [math.nan, "nan"]:
+        return "'NaN'"
+    elif lit in [float("inf"), "inf"]:
+        return "'inf'"
+    elif lit in [float("-inf"), "-inf"]:
+        return "'-inf'"
+    return str(lit)
+
+
+def literal_to_float(lit: str | int | float):
+    if lit in [float('inf'), 'inf']:
+        return "TO_DOUBLE('inf'::float)"
+    elif lit in [float('-inf'), '-inf']:
+        return "TO_DOUBLE('-inf'::float)"
+    return lit
+
+
+def is_float_type(arg):
+    return arg.type in ["fp32", "fp64"]
+
+
+
+def is_string_type(arg):
+    return (
+            arg.type in ["string", "timestamp", "timestamp_tz", "date", "time"]
+            and arg.value is not None
+    )
+
+
+def is_datetype(arg):
+    return type(arg) in [datetime.datetime, datetime.date, datetime.timedelta]
+
+
+class SnowflakeRunner(SqlCaseRunner):
+    def __init__(self, dialect):
+        super().__init__(dialect)
+        with open('bft/testers/snowflake/config.yaml', 'r') as file:
+            config = yaml.safe_load(file)
+            sf_config = config['snowflake']
+        print(
+            f"Connecting to {sf_config['account']} as {sf_config['username']}")
+        self.conn = connect(user=sf_config['username'],
+                            password=os.environ['SNOWSQL_PWD'],
+                            account=sf_config['account'],
+                            database=sf_config['database'],
+                            schema=sf_config['schema'],
+                            # host=sf_config['hostname'].get(""),
+                            # role=sf_config['role'].get(""),
+                            warehouse=sf_config['warehouse']
+                            )
+
+    def run_sql_case(self, case: Case, mapping: SqlMapping) -> SqlCaseResult:
```

Comment:
i'm guessing a large portion of this is duplicated between here and postgres. We should figure out how to factor out and share.

Reply:
I want to do this refactoring at a later point in time. Just want to see other changes that will be needed in snowflake tester.